### PR TITLE
baml-language: implement new PPIR

### PR DESCRIPTION
**Pull request by sxlijin**
## Release Notes

* **New Features**
  * Added stream type expansion functionality with support for type attributes
  * Type expressions now carry attributes for enhanced annotation capabilities

* **Tests**
  * Added comprehensive test fixtures for stream type coverage, including basic schemas, complex nested types, and edge case scenarios

### CODE REVIEW

File: baml_language/crates/bex_sap/src/sap_model/convert.rs [hunk 1]
```diff
@@ -146,37 +145,13 @@ pub fn convert(ty: &baml_type::Ty) -> Result<AnnotatedTy<'_, TypeName>, ConvertE
...
+    let in_progress = match attrs.sap_in_progress_never {
+        TyAttrValue::Set => Some(AttrLiteral::Never),
+        TyAttrValue::Unset => None,
     };
     Ok(TypeAnnotations {
-        in_progress: Some(convert_attr_literal(&attrs.sap_in_progress)?),
+        in_progress,
         parse_as: None,      // TODO: parse_as
         asserts: Vec::new(), // TODO: assertions
     })
```
[the remark this answers is not in the document]
Comment (sxlijin) [reply baml_language/crates/bex_sap/src/sap_model/convert.rs:175]: sap parsing will be handled in a followup

File: baml_language/crates/baml_compiler2_ppir/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        return PpirExpansionItems::new(db, Vec::new());
+    }
+
+    let cst = baml_compiler_parser::syntax_tree(db, file);
+    let (items, _) = ast::lower_file(&cst);
+
+    // Get HIR classification for the file's package (original types only)
+    let pkg_info = baml_compiler2_hir::file_package::file_package(db, file);
+    let pkg_id = PackageId::new(db, pkg_info.package);
+    let package_items = baml_compiler2_hir::package::package_items(db, pkg_id);
```
[the remark this answers is not in the document]
Comment (sxlijin) [reply baml_language/crates/baml_compiler2_ppir/src/lib.rs:148]: cross-pkg not handled
